### DISCUSSION TITLE
Possible to use nginx built-in ngx_http_auth_request module with dex?

### DISCUSSION THREAD

**Discussion by ghost** [dexidp/dex#3388] [Q&A]
Hi,

  Currently we use http auth to protect some web sites. Is it possible to configure Nginx to use its built-in auth_request module to proxy through to dex to authenticate/validate incoming users so they can access the site?

   Whenever I search for Nginx and OIDC I get lots of pages about Okta, Lua-resty-oidc (I believe Nginx is getting rid of Lua), and Nginx Plus.

   Any ideas or help getting pointed in the right direction?

Thanks!

**Comment by nabokihms** [comment]
With dex alone it is not possible, but you can achieve this using oauth2 proxy.

There is an old example, but you can get the idea
https://github.com/oauth2-proxy/oauth2-proxy/blob/master/contrib/local-environment/kubernetes/oauth2-proxy-example-full.yaml

**Comment by ghost** [comment]
Thanks @nabokihms ! Is it secure enough to run oauth2-proxy on the same server as dex or should they be on separate servers?

**Reply by nabokihms** [reply]
Secure enough to run on the same server AFAIK. It is better to improve pkce on the oauth2 proxy side to improve security.
